### PR TITLE
fix: Improve unit test coverage for product loading

### LINKED ISSUE #5927: Improve unit test coverage for product loading

### PR DESCRIPTION
**Pull request by dunqan**
Closes #5927

### CODE REVIEW

File: projects/core/src/product/services/product-loading.service.spec.ts [hunk 1]
```diff
@@ -62,103 +64,114 @@ describe('ProductLoadingService', () => {
...
         service
-          .get('testId', ['scope1', 'scope2'])
+          .get(code, ['scope1', 'scope2'])
           .pipe(take(3))
           .subscribe({
             next: res => {
               results.push(res);
             },
             complete: () => {
               expect(results).toEqual([
```
Comment (marlass) [inline projects/core/src/product/services/product-loading.service.spec.ts]: That's a nice pattern for multiple emissions tests :)

File: projects/core/src/product/services/product-loading.service.ts [hunk 2]
```diff
@@ -140,11 +148,19 @@ export class ProductLoadingService {
...
-            action.payload.code === productCode && action.meta.scope === scope
+          (
+            action:
+              | ProductActions.LoadProductSuccess
+              | ProductActions.LoadProductFail
+          ) =>
+            ((action.type === ProductActions.LOAD_PRODUCT_SUCCESS &&
+              action.payload.code === productCode) ||
+              (action.type === ProductActions.LOAD_PRODUCT_FAIL &&
+                action.payload === productCode)) &&
```
Comment (marlass) [inline projects/core/src/product/services/product-loading.service.ts]: I looked at the action and productCode is not in payload

Comment (dunqan) [reply projects/core/src/product/services/product-loading.service.ts]: Good catch. thanks!

File: projects/core/src/occ/services/loading-scopes.service.spec.ts [hunk 3]
```diff
@@ -56,4 +57,15 @@ describe('LoadingScopesService', () => {
...
     });
   });
+
+  describe('getMaxAge', () => {
+    it('should return maxAge in milliseconds', () => {
+      const result = service.getMaxAge('product', 'order');
+      expect(result).toEqual(60000);
+    });
+    it('should return 0 for not configured maxAge', () => {
+      const result = service.getMaxAge('product', 'details');
```
[the remark this answers is not in the document]
Comment (dunqan) [reply projects/core/src/occ/services/loading-scopes.service.spec.ts]: 👍

File: projects/core/src/product/services/product-loading.service.spec.ts (hunk 1 above)
Comment (dunqan) [reply projects/core/src/product/services/product-loading.service.spec.ts]: TBH I should use `scan` operator here :-) but idea would be the same.

**Pull request merged**